### PR TITLE
[framework] add Rax support

### PR DESCRIPTION
**Pull request by trongthanh**
Issue: #4797

## What we did

Add [Rax framework](https://github.com/alibaba/rax/) support

- [x] Implement `app/rax`
- [x] Implement generator `cli/generator/RAX` & framework detector
- [x] Implement `examples/rax-kitchen-sink`
- [x] Add and pass example unit tests
- [x] Add quick guide doc

## How to test

- Is this testable with Jest or Chromatic screenshots? Unsure, WIP
- Does this need a new example in the kitchen sink apps? Yes
- Does this need an update to the documentation? Yes

This PR is now beta, with all examples in rax-kitchen-sink working, with [know issues](https://github.com/raxland/storybook/issues). Any Rax developers interested in joining the effort, please contact me or leave a comment. Any contributions are welcomed.

### CODE REVIEW

**Comment by trongthanh** [comment]
I'm having an issue with storybook dev server not detect the framework for examples/rax-kitchen-sink. It is currently detected as @ storybook/vue project and provide wrong client render. If any maintainers can take a look at current PR and help us? I've found it very hard to debug the storybook CLI.

File: examples/rax-kitchen-sink/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "@storybook/addon-backgrounds": "5.0.0-beta.3",
+    "@storybook/addon-centered": "5.0.0-beta.3",
+    "@storybook/addon-events": "5.0.0-beta.3",
+    "@storybook/addon-info": "5.0.0-beta.3",
+    "@storybook/addon-jest": "5.0.0-beta.3",
+    "@storybook/addon-knobs": "5.0.0-beta.3",
+    "@storybook/addon-links": "5.0.0-beta.3",
+    "@storybook/addon-notes": "5.0.0-beta.3",
+    "@storybook/addon-options": "5.0.0-beta.3",
+    "@storybook/addon-storyshots": "5.0.0-beta.3",
```
Comment (shilman) [inline examples/rax-kitchen-sink/package.json]: Why no `@storybook/rax` dep?

Comment (trongthanh) [reply examples/rax-kitchen-sink/package.json]: Great catch. I've copied from other examples so lots of dependencies to compare and remove/add. I've got the rax-kitchen-sink working with basic button example for now. 

Thanks.

**Comment by ndelangen** [comment]
@trongthanh I'll take a look at this after the V5 release!

👍

**Comment by trongthanh** [comment]
Generator is done. But when I `sb init` a new storybook in a Rax project, I'm getting [this issue when using `rax-image` component](https://github.com/raxland/storybook/issues/6)

It is very hard to trace and debug this bug or may be I'm quite novice at how storybook works. Any helps are appreciated.

**Comment by shilman** [comment]
@ndelangen Wanna give him a hand?

**Comment by trongthanh** [comment]
I'm trying to iron out what's left for this PR in upcoming days (may be not in time for 4.1). Since there are so many changes from mainstream repo, please let me know which branch this PR should be merged into? Is it still `next`? @shilman

**Comment by shilman** [comment]
Hi @trongthanh glad to hear this is still going! New changes always come in to `next`, so no need to change your target branch. Linking Storybook is really a pain and it's never worked well for me (sadly).

My suggestion:
- Get things working on `examples/rax-kitchen-sink`, maybe add more tests to that
- Get this reviewed and merged
- I'll release it on `5.1.0-alpha.x` & you can test it on other projects & issue bugfix PRs

**Comment by trongthanh** [comment]
I have merged changes from latest 5.1.0-alpha.19.
Thanks to @hoanguyen311, we have added many working examples of addons in rax-kitchen-sink.
As @shilman suggested, I would like this PR reviewd and merged to current alpha release to test it in other projects.
In the mean time, we'll prepare documentations and a standalone kichen sink app to test this.

File: package.json [hunk 2]
```diff
@@ -42,7 +42,7 @@
     "publish:latest": "lerna publish --exact --concurrency 1 --force-publish",
     "publish:next": "npm run publish:latest -- --npm-tag=next",
     "repo-dirty-check": "node ./scripts/repo-dirty-check",
-    "start": "yarn --cwd examples/rax-kitchen-sink storybook",
+    "start": "yarn --cwd examples/official-storybook storybook",
```
Comment (shilman) [inline package.json]: Typo

**Comment by ndelangen** [comment]
Fixed merge conflict, Fixed linting issues, Fixes tests

**Comment by ndelangen** [comment]
I'll fix the merge-conflicts soon

**Review by ndelangen — APPROVED** [review]

**Comment by shilman** [comment]
@ndelangen can you fix yarn.lock so we can get the build working?

**Comment by trongthanh** [comment]
Please let us know how soon we can see this PR merged? We are pending to start a kitchen sink for a UI toolkit built with Rax.

**Comment by trongthanh** [comment]
Awesome! ❤️🎉

**Pull request merged**